src/ProjectOrigin.WalletSystem.Server/Services/REST/v1/ClaimsController.cs

```diff
@@ -117,13 +119,48 @@ public class ClaimsController : ControllerBase
             Quantity = request.Quantity,
         };
 
+        await unitOfWork.RequestStatusRepository.InsertRequestStatus(new Models.RequestStatus
+        {
+            RequestId = command.ClaimId,
+            Status = RequestStatusState.Pending
+        });
+
         await bus.Publish(command);
 
+        unitOfWork.Commit();
+
         return Accepted(new ClaimResponse()
         {
             ClaimRequestId = command.ClaimId,
         });
     }
+
+    /// <summary>
+    /// Gets status of specific claim.
+    /// </summary>
+    /// <param name="unitOfWork"></param>
+    /// <param name="claimRequestId">The ID of the claim request.</param>
+    /// <response code="200">The claim request status was found.</response>
+    /// <response code="401">If the user is not authenticated.</response>
+    /// <response code="404">If the claim specified is not found for the user.</response>
+    [HttpGet]
+    [Route("v1/claims/{claimRequestId}")]
+    [RequiredScope("po:claims:read")]
+    [ProducesResponseType(typeof(ClaimStatusResponse), StatusCodes.Status202Accepted)]
+    [ProducesResponseType(typeof(void), StatusCodes.Status401Unauthorized)]
+    [ProducesResponseType(typeof(void), StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<ClaimStatusResponse>> GetClaimStatus(
```

Comment:
Would prefer one RequestStatusController to avoid duplicated code. But if its complicated with Scopes then just resolve

Reply:
Done